app/saas.py

```diff
@@ -16,9 +16,13 @@ def home(*vargs):
     return render_template("display.html", meme_image=filename), 200
 
 
-@app.route("/<path:vargs>", methods=["GET"])
-def meme(vargs):
-    captions = vargs.split('/')
+@app.route("/q", methods=["GET"])
+def meme():
+    top = request.args.get('top')
```

Comment:
You will need to handle cases where either the top or the bottom captions aren't provided. You need to also store them in a dictionary to store which caption it is.

Reply:
Additionally, these changes will also affect utils.py, in the `generateImage()` function.

Reply:
Yeah sure, I'll make the changes and get back

Reply:
Solved in new commits